pkg/virt-controller/watch/ovm.go

```diff
@@ -295,6 +283,64 @@ func (c *OVMController) startStop(ovm *virtv1.OfflineVirtualMachine, vm *virtv1.
 	return nil
 }
 
+// setupVMfromOVM creates a VirtualMachine object from one OfflineVirtualMachine object.
+func (c *OVMController) setupVMFromOVM(ovm *virtv1.OfflineVirtualMachine) *virtv1.VirtualMachine {
+	basename := c.getVirtualMachineBaseName(ovm)
+
+	vm := virtv1.NewVMReferenceFromNameWithNS(ovm.ObjectMeta.Namespace, "")
+	vm.ObjectMeta = ovm.Spec.Template.ObjectMeta
+	vm.ObjectMeta.Name = basename
+	vm.ObjectMeta.GenerateName = basename
+	vm.Spec = ovm.Spec.Template.Spec
+
+	setupStableFirmwareUUID(ovm, vm)
+
+	t := true
+	// TODO check if vm labels exist, and when make sure that they match. For now just override them
+	vm.ObjectMeta.Labels = ovm.Spec.Template.ObjectMeta.Labels
+	vm.ObjectMeta.OwnerReferences = []v1.OwnerReference{v1.OwnerReference{
+		APIVersion:         virtv1.OfflineVirtualMachineGroupVersionKind.GroupVersion().String(),
+		Kind:               virtv1.OfflineVirtualMachineGroupVersionKind.Kind,
+		Name:               ovm.ObjectMeta.Name,
+		UID:                ovm.ObjectMeta.UID,
+		Controller:         &t,
+		BlockOwnerDeletion: &t,
+	}}
+
+	return vm
+}
+
+// TODO: make it const?
+// no special meaning, randomly generated on my box.
+// TODO: do we want to use another constants? see examples in RFC4122
+var _uuid_ns = uuid.Parse("6a1a24a1-4061-4607-8bf4-a3963d0c5895")
```

Comment:
At the very least, this magic string could be a const. It appears this is being used to salt a hash?

Reply:
@stu-gott this is the namespace I was referring to.
"namespace" is meant in this context: https://tools.ietf.org/html/rfc4122#page-13

I agree it should be a const, will make it so in the next upload.

Reply:
Well, actually there is no trivial way to make this expression constant, because it can't be evaluated at compile time (see for more details this: https://stackoverflow.com/questions/13137463/declare-a-constant-array)

So far I just renamed to something more go-friendly, like "firmwareUUIDNs"

Reply:
Oh sorry for not being clear. I meant something along the lines of:

```
const magicUUID = "6a1a24a1-4061-4607-8bf4-a3963d0c5895"
var firmwareUUIDns = uuid.Parse(magicUUID)
```

Totally understand that "parse" isn't a compile time thing.

Reply:
Right, done.